### PR TITLE
Add support for loading supported features from share/swift/features.json

### PR DESCRIPTION
**Pull request by nkcsgexi**

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please test

**Comment by nkcsgexi** [comment]
@swift-ci please test

**Comment by nkcsgexi** [comment]
@swift-ci please test

File: Sources/SwiftDriver/ExplicitModuleBuilds/ModuleDependencyScanning.swift [hunk 1]
```diff
@@ -344,4 +344,24 @@ public extension Driver {
       .parentDirectory // bin
       .parentDirectory // toolchain root
   }
+
+  static func computeSupportedCompilerFeatures(of toolchain: Toolchain,
```
Comment (artemcm) [inline Sources/SwiftDriver/ExplicitModuleBuilds/ModuleDependencyScanning.swift:348]: Could we please move this to be alongside the equivalent supported-flags query in `EmitSupportedFeaturesJob.swift`?

**Review by artemcm — APPROVED** [review]
Cool!

File: Sources/SwiftDriver/ExplicitModuleBuilds/ModuleDependencyScanning.swift (hunk 1 above)
Comment (nkcsgexi) [reply Sources/SwiftDriver/ExplicitModuleBuilds/ModuleDependencyScanning.swift]: Fixed!

**Comment by nkcsgexi** [comment]
@swift-ci please test

**Pull request merged**